NitroxServer-Subnautica/Communication/Packets/Processors/RocketLaunchProcessor.cs

```diff
@@ -0,0 +1,44 @@
+﻿using NitroxModel.DataStructures.Util;
+using NitroxModel_Subnautica.DataStructures.GameLogic;
+using NitroxModel_Subnautica.Packets;
+using NitroxServer.Communication.Packets.Processors.Abstract;
+using NitroxServer.GameLogic;
+using NitroxServer.GameLogic.Vehicles;
+
+namespace NitroxServer_Subnautica.Communication.Packets.Processors;
+
+public class RocketLaunchProcessor : AuthenticatedPacketProcessor<RocketLaunch>
+{
+    private readonly VehicleManager vehicleManager;
+    private readonly PlayerManager playerManager;
+
+    public RocketLaunchProcessor(VehicleManager vehicleManager, PlayerManager playerManager)
+    {
+        this.vehicleManager = vehicleManager;
+        this.playerManager = playerManager;
+    }
+
+    public override void Process(RocketLaunch packet, NitroxServer.Player player)
+    {
+        Optional<NeptuneRocketModel> opRocket = vehicleManager.GetVehicleModel<NeptuneRocketModel>(packet.RocketId);
+
+        if (opRocket.HasValue)
+        {
+            // 5 is the value of RocketPreflightCheckManager.totalPreflightChecks
+            if (opRocket.Value.PreflightChecks.Count == 5)
+            {
+                // Tell every player to launch rocket
+                playerManager.SendPacketToAllPlayers(packet);
+            }
+            else
+            {
+                // Resync the player's preflight checks
+                player.SendPacket(new RocketResync(packet.RocketId, opRocket.Value.PreflightChecks.ToList()));
```

Comment:
In one case we send packet to everyone, and in the other case to a specific person ?